### ISSUE TITLE
[BUG] Does not work with pnpm

### ISSUE THREAD

**Issue by mczapkowicz** [callstack/reassure#277]
**Describe the bug**
When using **pnpm** as a package manager reassure binary cannot be found.
After running command **pnpm reassure** i get:

` ERR_PNPM_RECURSIVE_EXEC_FIRST_FAIL  not found: reassure`

I noticed that reassure binary is missing in the _node_modules/.bin_ folder.

**To Reproduce**
1. Create react project with _create-react-app_
2. Add **reassure** to the project using **pnpm add reassure -D**
3. run **pnpm reassure**

**Comment by mdjastrzebski** [comment]
@mczapkowicz you are right, we did not test pnmp compatibility. That the issue you are facing should be relatively easy to work around: you can set [`TEST_RUNNER_PATH`](https://github.com/callstack/reassure#environmental-variables) env variable to proper `reassure` CLI path.

**Comment by mczapkowicz** [comment]
Yes, the error is exactly the same:
` ERR_PNPM_RECURSIVE_EXEC_FIRST_FAIL  not found: reassure`

**Comment by mdjastrzebski** [comment]
I think we've confusing two things here. One is TEST_RUNNER_PATH which should have path to test runner, e.g. Jest
While you issue seems to be about executing Reassure CLI itself. Since this is a simple node executable you should be able to run it using:
* `node node_modules/reassure/node_modules/.bin/reassure`
* or: `node node_modules/.bin/reassure`
* or  `./node_modules/reassure/node_modules/.bin/reassure`

**Comment by mczapkowicz** [comment]
- `TEST_RUNNER_PATH=node_modules/.bin/jest node node_modules/reassure/node_modules/.bin/reassure
`

gives me:
```
 /Users/user/Desktop/test-project/node_modules/.pnpm/reassure@0.6.0_react@18.2.0/node_modules/reassure/node_modules/.bin/reassure:2
basedir=$(dirname "$(echo "$0" | sed -e 's,\\,/,g')")
          ^^^^^^^

SyntaxError: missing ) after argument list
    at Object.compileFunction (node:vm:352:18)
    at wrapSafe (node:internal/modules/cjs/loader:1033:15)
    at Module._compile (node:internal/modules/cjs/loader:1069:27)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1159:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:77:12)
    at node:internal/main/run_main_module:17:47
```

- `node node_modules/.bin/reassure`, gives: 
`Cannot find module '/Users/user/Desktop/test-project/node_modules/.bin/reassure`
 which makes sense because, like I said there is no reassure in **node_modules/.bin** folder
- `./node_modules/reassure/node_modules/.bin/reassure` gives

```
node:internal/modules/cjs/loader:936
  throw err;
  ^

Error: Cannot find module '/Users/user/Desktop/test-project/@callstack+reassure-cli@0.5.0/node_modules/@callstack/reassure-cli/lib/commonjs/bin.js'
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)
    at Function.Module._load (node:internal/modules/cjs/loader:778:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:77:12)
    at node:internal/main/run_main_module:17:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}
```

**Comment by mdjastrzebski** [comment]
@mczapkowicz:
* what OS are you using? Another used reported the first error (`=$(dirname "$(echo "$0" | sed -e 's,\\,/,g')")`) for Windows env.
* could you try to use it with `yarn` or `npm`? I wonder if this is more OS-specific of package manager specific?

**Comment by mczapkowicz** [comment]
I'm using macOS. Everything works fine with yarn and npm so I belive this is package manager specific.

**Comment by mdjastrzebski** [comment]
@mczapkowicz do you have some available time and would like to bring PNMP support to Reassure? 😀

The issue here seems to be specific to  how PNPM works then the actual Reassure code.

**Comment by mczapkowicz** [comment]
I will take a look at this.

**Comment by Zenor27** [comment]
Same problem here using on macOS, seems related to pnpm indeed.

**Comment by gdagundaridze** [comment]
For anyone that may benefit, i was able to make this work with `pnpm` in a slightly ugly but temporarily acceptable way like this:
```TEST_RUNNER_PATH=node_modules/jest/bin/jest ../../node_modules/.pnpm/reassure@0.7.2_react@18.2.0/node_modules/reassure/node_modules/.bin/reassure```

Will try to submit a PR so that hacking like this is not required for pnpm

**Comment by d7sd6u** [comment]
To properly fix this problem it is better to use `.pnpmfile.cjs`. Create that file with the following contents (or modify your existing file accordingly):

```typescript
module.exports = {
  hooks: {
    readPackage: (pkg) => {
      if (pkg.name === "reassure") {
        pkg.bin = {
          reassure: "./node_modules/.bin/reassure",
        }
      }
      if (pkg.name === "@callstack/reassure-measure") {
        pkg.peerDependencies["@testing-library/react"] = "*"; // change to /react-native if you are using it instead
      }
    }
  }
}
```

This hook modifies `package.json` of the packages on-the-fly when pnpm resolves dependencies. It adds missing dependencies and `.bin` field, because currently they are not correctly set (as they always should be, because packages shouldn't rely on fantom dependecies!).

The PR therefore should be straightforward: just add missing dependencies and `.bin` field to the actual `package.json` files...

**Comment by V3RON** [comment]
It seems the issue has been resolved. I successfully installed and ran Reassure in a pnpm + Vite + ts-jest environment, and it worked perfectly 👌 

Can someone involved confirm this?

**Comment by MilanBarande** [comment]
Yes works just fine with pnpm, this issue should be closed
